ambry-messageformat/src/test/java/com/github/ambry/messageformat/BlobStoreHardDeleteTest.java

```diff
@@ -396,6 +424,49 @@ public void blobStoreHardDeleteTestBlobV2Mixed() throws MessageFormatException,
     blobStoreHardDeleteTestUtil(blobVersions, blobTypes);
   }
 
+  @Test
+  public void blobStoreHardDeleteTestBlobV3Mixed() throws MessageFormatException, IOException {
+
+    short[] blobVersions = new short[5];
+    BlobType[] blobTypes = new BlobType[5];
+
+    blobVersions[0] = MessageFormatRecord.Blob_Version_V1;
+    blobTypes[0] = BlobType.DataBlob;
+
+    blobVersions[1] = MessageFormatRecord.Blob_Version_V2;
+    blobTypes[1] = BlobType.MetadataBlob;
+
+    blobVersions[2] = MessageFormatRecord.Blob_Version_V2;
+    blobTypes[2] = BlobType.DataBlob;
+
+    blobVersions[3] = MessageFormatRecord.Blob_Version_V3;
+    blobTypes[3] = BlobType.MetadataBlob;
+
+    blobVersions[4] = MessageFormatRecord.Blob_Version_V3;
+    blobTypes[4] = BlobType.DataBlob;
+
+    blobStoreHardDeleteTestUtil(blobVersions, blobTypes);
+
+    blobVersions = new short[5];
+    blobTypes = new BlobType[5];
+    blobVersions[0] = MessageFormatRecord.Blob_Version_V1;
+    blobTypes[0] = BlobType.DataBlob;
+
+    blobVersions[1] = MessageFormatRecord.Blob_Version_V2;
+    blobTypes[1] = BlobType.DataBlob;
+
+    blobVersions[2] = MessageFormatRecord.Blob_Version_V2;
+    blobTypes[2] = BlobType.MetadataBlob;
+
+    blobVersions[3] = MessageFormatRecord.Blob_Version_V3;
+    blobTypes[3] = BlobType.DataBlob;
+
+    blobVersions[4] = MessageFormatRecord.Blob_Version_V3;
+    blobTypes[4] = BlobType.MetadataBlob;
+
+    blobStoreHardDeleteTestUtil(blobVersions, blobTypes);
```

Comment:
repeated code ?

Reply:
use a loop or call some helper twice

Reply:
They look very similar, almost identical, but these 2 sets are actually testing different sequences.  
The 1st sequence is Data, Metadata, Data, Metadata, Data.
The 2nd sequence is Data, Data, Metadata, Data, Metadata.

Because this is unit test, I wouldn't be so concern about it.

Reply:
ok please add this in the comments otherwise at first glance they look the same

Reply:
added.